### ISSUE TITLE
[question] WriteToFile / Magicksave async

### ISSUE THREAD

**Issue by lofcz** [kleisauke/net-vips#110]
From my current understanding, while calling one of these methods to save an image, the current thread hangs instead of doing someting useful while an image is being saved. Would it be possible to introduce async overloads of these methods?

**Comment by kleisauke** [comment]
Indeed, the `WriteTo{File,Buffer,Target,Stream}` and `*save{Buffer,Target,Stream}` operations are currently missing async overloads. As for the implementation of this, I think it only makes sense to use async for operations that will trigger a pixel loop, since most NetVips operations are instant.

Let's tag this as an enhancement. Very happy to accept a PR, if you're able.
